### PR TITLE
Perform a small optimization of HDRP shader compile times [On Hold]

### PR DESCRIPTION
**Pull request by EvgeniiG**
### Purpose of this PR
- Add 'MaterialInstanceFlags' which are used for conversion of Shader Features into uniform branches
- Remove '_PIXEL_DISPLACEMENT_LOCK_OBJECT_SCALE' and '_DISPLACEMENT_LOCK_TILING_SCALE' as an example

## Testing status
**Katana Tests**: soon (tm)

### Comments to reviewers
Upgrading old materials needs to be tested.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Lighting.hlsl [hunk 1]
```diff
@@ -7,34 +7,7 @@
...
-#endif
-
-#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Material/Material.hlsl" // Depends on LightLoopDef and shadows
-
-// Volumetrics have their own light loop.
-#ifndef UNITY_MATERIAL_VOLUMETRIC
-    // LightLoop use evaluation BSDF function for light type define in Material.hlsl
-    #if defined(LIGHTLOOP_SINGLE_PASS) || defined(LIGHTLOOP_TILE_PASS)
-        #if HDRP_MATERIAL_TYPE_SIMPLE
-            #include "Packages/com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/SimpleLightLoop.hlsl"
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Lighting.hlsl]: Be careful, VFX graph will break. We need to update VFX graph to it use the SimpleLightLoop.hlsl file correctly

**Comment by EvgeniiG** [comment]
Should I manually upgrade test scenes? Do we want this PR at all?

File: com.unity.render-pipelines.high-definition/Runtime/Material/Material.hlsl [hunk 2]
```diff
@@ -9,6 +9,8 @@
 #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/CommonMaterial.hlsl"
 #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/EntityLighting.hlsl"
 #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/ImageBasedLighting.hlsl"
+
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Material/MaterialInstanceFlags.cs.hlsl"
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Material/Material.hlsl]: I don't like the fact that we have something specific to Lit/layeredlit shader, as something general. 
(Note: our new politics is to have everything as master node except the lit, layeredlit and AxF shader, so material instance is fully specific to lit shader).

**Comment by sebastienlagarde** [comment]
>Should I manually upgrade test scenes? Do we want this PR at all?

Let's discuss about it. This PR is a start for the optimization of lit/layeredLit variant issue (less variant mean also more SRP batcher friendly). I think we need to go further than this (but not now) and will require to develop a strong upgrade material process to not mess with current project. 

Here we talk about upgrading: ScriptableRenderPipelineData, Fontainebleau, VFX demo, FPS sample game, graphic test from HDRP_Test, and have a solid upgrader for external project

**Comment by sebastienlagarde** [comment]
Closing this PR as we can't use it in current state anyway. Too old now.

**Pull request closed without merging**